src/components/Map.js

```diff
@@ -95,15 +96,21 @@ function Map() {
 
     map.current.once("idle", () => {
       if (watercourseLinkId) {
-        showWatercourseLink(watercourseLinkId, map);
-        setShowSearch(false);
+        showWatercourseLink(watercourseLinkId, map)
+          .then(() => {
+            setShowSearch(false);
+          })
+          .catch((error) => {
+            setSearchError(error);
+            setShowSearch(true);
+          });
       }
     });
   }, [watercourseLinkId]);
 
   return (
     <>
-      <Search map={map} initialShow={showSearch} />
+      <Search map={map} initialShow={showSearch} initialError={searchError} />
```

Comment:
slightly worried that it's getting more complex/messy with more things being passed to Search - but maybe it's okay and I'm overthinking? 😬

Reply:
I think it's fine.. I've definitely seen much longer props lists in React apps! There are nicer ways to handle state in React (like some central state management type of thing like re-frame does), but none are set up in this project. If we were doing it "for real" (and still using React for some reason), we'd probably do that. But for now passing props around is fine, if not ideal.